client/components/plans/plan-actions/index.jsx

```diff
@@ -95,57 +112,77 @@ module.exports = React.createClass( {
 			<div>
 				<button
 					className="button is-primary plan-actions__upgrade-button"
-					onClick={ this.handleAddToCart.bind( null, this.cartItem( { isFreeTrial: false } ), 'button' ) }>
+					disabled={ this.props.isSubmitting }
+					onClick={ this.handleAddPlanToCart }>
 					{ label }
 				</button>
 			</div>
 		);
 	},
 
-	recordStartFreeTrialClick: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Start Free Trial Button', 'Product ID', cartItem.product_id );
-	},
-
-	recordUpgradeNowClick: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link', 'Product ID', cartItem.product_id );
+	recordStartFreeTrialClick: function() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Start Free Trial Button', 'Product ID', this.props.plan.product_id );
 	},
 
-	recordUpgradeNowButton: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Button', 'Product ID', cartItem.product_id );
+	recordUpgradeNowClick: function() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link', 'Product ID', this.props.plan.product_id );
 	},
 
-	recordUpgradeTrialNowClick: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link For Trial', 'Product ID', cartItem.product_id );
+	recordUpgradeNowButton: function() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Button', 'Product ID', this.props.plan.product_id );
 	},
 
-	handleAddToCart: function( cartItem, actionType, event ) {
+	handleAddPlanToCart: function( event ) {
 		if ( event ) {
 			event.preventDefault();
 		}
 
-		if ( cartItem && actionType ) {
-			if ( actionType === 'button' ) {
-				if ( cartItem.free_trial ) {
-					this.recordStartFreeTrialClick( cartItem );
-				}
-				if ( ! cartItem.free_trial ) {
-					this.recordUpgradeNowButton( cartItem );
-				}
-			}
+		if ( this.props.isSubmitting ) {
+			return;
+		}
 
-			if ( actionType === 'link' ) {
-				if ( cartItem.free_trial ) {
-					this.recordUpgradeTrialNowClick( cartItem );
-				}
-				if ( ! cartItem.free_trial ) {
-					this.recordUpgradeNowClick( cartItem );
-				}
-			}
+		const actionType = event.target.tagName === 'A' ? 'link' : 'button';
+
+		if ( 'link' === actionType ) {
+			this.recordUpgradeNowClick();
+		} else {
+			this.recordUpgradeNowButton();
 		}
 
+		const cartItem = cartItems.getItemForPlan( this.props.plan );
+
 		if ( this.props.onSelectPlan ) {
-			this.props.onSelectPlan( cartItem );
+			return this.props.onSelectPlan( cartItem );
 		}
+
+		upgradesActions.addItem( cartItem );
+
+		page( '/checkout/' + this.props.site.slug );
+	},
+
+	handleStartFreeTrial: function( event ) {
```

Comment:
@Tug This is super minor, but I think we should only prefix methods with `handle` that relate to the event to which they are bound, e.g. `handleClickStartTrial`. Ideally, we have methods that are simple enough that we can just set event handlers like `onClick={ this.startTrial }`, but this method also handles recording the click, so I think using a handle-prefixed method is appropriate.